ocaml/libs/http-lib/server_io.ml

```diff
@@ -26,15 +26,83 @@ type handler = {
   ; lock: Semaphore.Counting.t
 }
 
+module WorkerPool = struct
+  type task = unit -> unit
+
+  type t = {
+      queue: task Queue.t
+    ; size: int
+    ; lock: Mutex.t
+    ; cond_non_empty: Condition.t
+  }
+
+  let worker p =
+    let next () =
+      Mutex.lock p.lock ;
+      while Queue.is_empty p.queue do
+        Condition.wait p.cond_non_empty p.lock
+      done ;
+      let task = Queue.pop p.queue in
+      Mutex.unlock p.lock ;
+      ignore (task ())
```

Comment:
Also should we catch exceptions here? If a 'task' raises then the entire worker thread is dead. That may be a neat way to shut down the worker thread, but buggy API calls can then easily take it down.

Reply:
Fun.protect only wraps it with a finally that gets called on exception, but the exception is reraised and escapes.
With the current thread based handler that is fine: the thread dies, but the thread only handled 1 API call, so that is fine.

In this case if a worker dies then eventually we'll be out of workers and get stuck.

Would be useful to wrap calls to `task` with `try / except`, or with Debug.log_and_ignore_exn.